### PR TITLE
store: Cleaned up API for test/benchmark purposes.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
This is so we can use our structs for reliable go benchmarks from outside package (e.g to integrate multiple things): https://gist.github.com/bwplotka/cbcbbcd1802181b7785da11dcc0f5cfd#

File: test/e2e/e2ethanos/services.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 	"github.com/thanos-io/thanos/pkg/receive"
 )
 
-const logLevel = "info"
+const infologLevel = "info"
```
Comment (prmsrswt) [inline test/e2e/e2ethanos/services.go:30]: Just a small nit.

```suggestion
const infoLogLevel = "info"
```

File: test/e2e/e2ethanos/services.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func NewQuerier(sharedDir, name string, storeAddresses, fileSDStoreAddresses, ru
 		"--http-address":          ":8080",
 		"--query.replica-label":   replicaLabel,
 		"--store.sd-dns-interval": "5s",
-		"--log.level":             logLevel,
+		"--log.level":             "debug", // remove later.
```
Comment (prmsrswt) [inline test/e2e/e2ethanos/services.go:127]: Need to flip this back to `infoLogLevel`.

File: pkg/api/query/v1.go [hunk 3]
```diff
@@ -307,6 +308,7 @@ func (qapi *QueryAPI) query(r *http.Request) (interface{}, []error, *api.ApiErro
 }
 
 func (qapi *QueryAPI) queryRange(r *http.Request) (interface{}, []error, *api.ApiError) {
+	reqTime := time.Now()
```
Comment (prmsrswt) [inline pkg/api/query/v1.go:311]: This was added for debugging I guess.

File: pkg/store/bucket_test.go [hunk 4]
```diff
@@ -2037,7 +2034,7 @@ func setupStoreForHintsTest(t *testing.T) (testutil.TB, *BucketStore, []*storepb
 		tmpDir,
 		indexCache,
 		nil,
-		1000000,
+		nil,
```
Comment (prmsrswt) [inline pkg/store/bucket_test.go:2037]: Don't we need a `chunkPool` with `maxTotal = 100000` here?

**Review by prmsrswt — COMMENTED** [review]
Looks good to me, just a few nits.

File: pkg/block/fetcher.go [hunk 5]
```diff
@@ -184,6 +184,12 @@ func NewBaseFetcher(logger log.Logger, concurrency int, bkt objstore.Instrumente
 	}, nil
 }
 
+// NewRawMetaFetcher returns basic meta fetcher without proper handling for eventual consistent backends or partial uploads.
+// NOTE: Not suitable to use in production.
+func NewRawMetaFetcher(logger log.Logger, bkt objstore.InstrumentedBucketReader) (*MetaFetcher, error) {
```
Comment (yeya24) [inline pkg/block/fetcher.go:189]: I think it is not a good practice to have a public method that is only for testing purposes? I only find this method used in unit tests, then is it better to move this method to that package?

File: pkg/store/bucket.go [hunk 6]
```diff
@@ -282,14 +283,33 @@ type BucketStore struct {
...
+}
+
+type noopGate struct{}
+
+func (noopGate) Start(context.Context) error { return nil }
+func (noopGate) Done()                       {}
+
 // NewBucketStore creates a new bucket backed store that implements the store API against
 // an object store bucket. It is optimized to work against high latency backends.
+// TODO(bwplotka): Move to config at this point (:
```
Comment (yeya24) [inline pkg/store/bucket.go:312]: +1

**Comment by bwplotka** [comment]
Thanks for reviews, will update!

File: pkg/block/fetcher.go (hunk 5 above)
Comment (bwplotka) [reply pkg/block/fetcher.go:191]: Do you have any suggestion, if I want to reuse this across tests?

File: pkg/pool/pool.go [hunk 7]
```diff
@@ -9,15 +9,28 @@ import (
 	"github.com/pkg/errors"
 )
 
-type BytesPool interface {
+// Bytes is a pool of bytes that can be reused.
+type Bytes interface {
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply pkg/pool/pool.go:13]: `BytesPool` it's too clear, we already have `pool` as a package name so it's known it's pool. That's kind of idiomatic. I would vote to leverage pool pacakge WDYT?

File: test/e2e/e2ethanos/services.go (hunk 1 above)
Comment (bwplotka) [reply test/e2e/e2ethanos/services.go:30]: Thanks for catching!

**Comment by bwplotka** [comment]
Rebased!

**Comment by bwplotka** [comment]
PTAL @yeya24 @prmsrswt

File: pkg/api/query/v1.go [hunk 8]
```diff
@@ -350,7 +350,7 @@ func (qapi *QueryAPI) queryRange(r *http.Request) (interface{}, []error, *api.Ap
 	}
 
 	step, apiErr := qapi.parseStep(r, qapi.defaultRangeQueryStep, int64(end.Sub(start)/time.Second))
-	if err != nil {
+	if apiErr != nil {
```
Comment (yeya24) [inline pkg/api/query/v1.go:353]: Amazing. How did you catch this

**Review by yeya24 — APPROVED** [review]
LGTM

File: pkg/api/query/v1.go (hunk 8 above)
Comment (bwplotka) [reply pkg/api/query/v1.go:353]: Laser eyes

**Pull request merged**